Legal Compliance and More

Wans-G opened the pull request:
## Changes

### 1. Added Legal Compliance Modal to the First-Login Page
- **Feature**:
  - A modal displaying legal compliance information is shown on the user's first login.
  - A cookie records whether the modal has been viewed, ensuring it doesn’t appear again unless the user switches to private browsing mode.

### 2. Terms Modal on the Registration Page
- **Feature**:
  - Before registering, users can click a link to view a modal displaying the terms of service.
- **Details**:
  - A "View Terms" link is added to the registration page.
  - A new modal component is created to display terms of service.

### 3. Music Activation Button
- **Feature**:
  - Added a button on the landing page to activate background music.
  - For Google login users, a modal automatically pops up to display exclusive discount information.
- **Details**:
  - A music activation button added to `landing.html.haml`.
  - JavaScript checks if the user logged in via Google and triggers the discount modal if applicable.
  - The modal includes a "Go to Store" button linking to `/store_items`.

### 4. Added Website Icon and Fixed Favicon Error
- **Feature**:
  - Added a favicon to enhance the website’s appearance.
  - Resolved the recurring error in the terminal:  
    `ActionController::RoutingError (No route matches [GET] "/favicon.ico")`.

### 5. Easter Egg
  - Added a small Easter egg to surprise users. 

### 6. Testings
- Rspec and Cucumber for Legal Compliance Page

bjohnson66 reviewed (changes requested):
See comments on legal parts. 
Also please take log file out of changeset
----
57 scenarios (57 passed)
570 steps (570 passed)
Line Coverage: 93.61% (498 / 532)
Nice!
----


I see two scroll bars, can we get it to one?
![image](https://github.com/user-attachments/assets/4ee8187f-7c7e-4363-8cd0-e39144720f97)



That was funny., I like it. Make sure we dont forget that is in there:
![image](https://github.com/user-attachments/assets/115346b1-13bd-4ce8-9d59-1761d9348cfd)

pretty:
![image](https://github.com/user-attachments/assets/a5ddd2a4-8340-4896-8d4e-b05ae8f8484b)

If i click disagree from legal page, i get this:
![image](https://github.com/user-attachments/assets/c6660efc-5732-4869-9224-06a0d2d18d7d)

Agree button seems to work for both pages. :)

Wans-G commented:
@bjohnson66 I updated the change and removed the test.log.
I tried the partial HTML, and it worked. But the problem is that the agree/disagree buttons are part of the same modal as the legal content. We have one page that needs the two buttons, and the other doesn't(register page).  I think that would make a redesign of the page and take more effort to change the structure. I will keep the old version for now. If we finish everything early, I can go back and redesign these. But it's good to know, so in the future I can consider reusing components!

bjohnson66 commented on app/views/users/new.html.haml:
```diff
@@ -10,22 +10,118 @@
+      Welcome to the Legal Compliance Notice for Shards of the Grid. Please read the following information carefully to understand our Terms of Service and Privacy Policy.
+    %p
+    %p
+    %h2 Terms of Service
```
Its fine

bjohnson66 reviewed (approved):
Line Coverage: 93.61% (498 / 532)
scrolling is fixed
